contracts/zero-ex/contracts/src/transformers/FillQuoteTransformer.sol

```diff
@@ -388,31 +392,105 @@ contract FillQuoteTransformer is
         private
         returns (FillOrderResults memory results)
     {
-        // Track changes in the maker token balance.
-        uint256 initialMakerTokenBalance = makerToken.balanceOf(address(this));
-        try
-            exchange.fillOrder
-                {value: protocolFee}
-                (order, takerAssetFillAmount, signature)
-            returns (IExchange.FillResults memory fillResults)
-        {
-            // Update maker quantity based on changes in token balances.
-            results.makerTokenBoughtAmount = makerToken.balanceOf(address(this))
-                .safeSub(initialMakerTokenBalance);
-            // We can trust the other fill result quantities.
-            results.protocolFeePaid = fillResults.protocolFeePaid;
-            results.takerTokenSoldAmount = fillResults.takerAssetFilledAmount;
-            // If the taker fee is payable in the taker asset, include the
-            // taker fee in the total amount sold.
-            if (isTakerFeeInTakerToken) {
-                results.takerTokenSoldAmount =
-                    results.takerTokenSoldAmount.safeAdd(fillResults.takerFeePaid);
+        bytes4 makerAssetProxyId = LibBytesV06.readBytes4(order.makerAssetData, 0);
+        // If it is a Bridge order we fill this directly
+        // rather than filling via 0x Exchange
+        if (makerAssetProxyId == ERC20_BRIDGE_PROXY_ID) {
+            try
+                this.fillBridgeOrder(order,
+                    signature,
+                    takerAssetFillAmount,
+                    makerToken)
```

Comment:
we intentionally ignore fee related settlement, so no need to adjust `takerTokenSoldAmount`